WordPress/gutenberg
# Gallery block: Remove warning notice about mobile version required

## Description
Fixes: https://github.com/Automattic/wp-calypso/issues/59560

In a small number of instances, the one-time notice about the Gallery mobile version issue displays multiple times every time a gallery is added or edited. It is not possible to replicate this, so difficult to work out how to prevent it. 

It appears that for a very small number of users the value that toggles this message off is not sent to the store, but with no easy way to replicate it is not possible to work out the cause of that or a workaround.

Given that the number of mobile users on < 18.2 will should now be reduced considerably since this message was added, combined with the risk of this issue affecting a large number of users if it goes out in WP5.9 this PR removes the message completely.

## To test

- Check out this PR to local dev env, and clear the local storage for the site
- Add a new gallery and add images to it, and make sure no warning message about mobile app versions displays
- Make sure everything else in gallery works as expected

## Screenshots
Before:
![message-before](https://user-images.githubusercontent.com/3629020/148857039-86a36144-955a-4984-a820-a77c1e770dd6.gif)

After:
![message-after](https://user-images.githubusercontent.com/3629020/148857049-794bfc1f-d0d5-4a69-8870-2dbcee8d5820.gif)
